saleor/payment/gateways/stripe/tests/test_webhooks.py

```diff
@@ -1170,6 +1170,50 @@ def test_handle_refund_already_processed(
     assert payment.captured_amount == payment.total - Decimal("10")
 
 
+@patch("saleor.payment.gateways.stripe.webhooks.stripe.Charge.retrieve")
+def test_handle_refund_missing_refunds(
+    charge_retrieve, stripe_plugin, payment_stripe_for_order, channel_USD
+):
+    # given
+    payment = payment_stripe_for_order
+    payment.captured_amount = payment.total
+    payment.save()
+    payment.transactions.create(
+        is_success=True,
+        action_required=True,
+        kind=TransactionKind.CAPTURE,
+        amount=payment.total,
+        currency=payment.currency,
+        token="ABC",
+        gateway_response={},
+    )
+    plugin = stripe_plugin()
+
+    charge = StripeObject()
+    charge["payment_intent"] = "ABC"
+
+    refund = StripeObject(id="refund_id")
+    refund["amount"] = price_to_minor_unit(payment.total, payment.currency)
+    refund["currency"] = payment.currency
+    refund["last_response"] = None
+
+    charge_retrieve_obj = charge
+    charge_retrieve_obj["refunds"] = StripeObject()
```

Comment:
IMO we should add one more test when there are no `refunds` as currently we are not testing the part of the code that has been changed

Reply:
Discussed with @fowczarek - test cases for this will be delivered in separate PR after Krzysieks vacations. Issue has been created to keep track of it: https://github.com/saleor/saleor/issues/15028